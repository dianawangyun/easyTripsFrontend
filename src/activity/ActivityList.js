import React, { useState } from "react";
import ActivityCard from "./ActivityCard";
import LoadingSpinner from "../common/LoadingSpinner";

/* map activities and render ActivityCard. */
function ActivityList({ trip, deleteActivity }) {
    /* If isEditing, then operations buttons would hide. */
    const [isEditing, setIsEditing] = useState(false);

    if (!trip) return <LoadingSpinner />;

    return (
        <div>
            {trip.activities.length ? (
                <ul className="list-group">
                    {trip.activities.map((a) => (
                        <ActivityCard
                            key={a.activityId}
                            activityId={a.activityId}
                            activityName={a.activityName}
                            description={a.description}
                            location={a.location}
                            latitude={a.latitude}
                            longitude={a.longitude}
                            startTime={a.startTime}
                            endTime={a.endTime}
                            comment={a.comment}
                            deleteActivity={deleteActivity}
                            setIsEditing={setIsEditing}
                            isEditing={isEditing}
                        />
                    ))}
                </ul>
            ) : (
                <p className="lead">
                    No Activities in this trip. Let's Add some!
                </p>
            )}
        </div>
    );
}
export default ActivityList;
