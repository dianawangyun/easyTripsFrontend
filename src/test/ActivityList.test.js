import React from "react";
import { render } from "@testing-library/react";
import ActivityList from "../activity/ActivityList";

it("renders without crashing", function () {
    render(<ActivityList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<ActivityList trip={{ activities: [] }} />);
    expect(asFragment()).toMatchSnapshot();
});

it("should show no activity", function () {
    const { queryByText } = render(<ActivityList trip={{ activities: [] }} />);
    expect(
        queryByText("No Activities in this trip. Let's Add some!")
    ).toBeInTheDocument();
});
