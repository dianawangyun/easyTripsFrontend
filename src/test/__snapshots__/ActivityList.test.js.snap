// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      class="lead"
    >
      No Activities in this trip. Let's Add some!
    </p>
  </div>
</DocumentFragment>
`;
